src/components/Carousel/FirstVedioCarousel.jsx

```diff
@@ -0,0 +1,49 @@
+import "./vedioCarousel.css";
+import { Carousel } from "./Carousel";
+import { firstVedioCarouselData } from "../../utils/utilityArrays";
+import { useNavigate } from "react-router-dom";
+import { useFilterData } from "../../contexts/filterData-context";
+
+const FirstVedioCarousel = () => {
+
+    const navigate = useNavigate();
+    const { dispatchOfProductsBeingShown } = useFilterData();
+
+    const handleFirstVedioCarouselClick = (eachVedioData) => {
+        dispatchOfProductsBeingShown({ type: "clear" });
+
+        if(eachVedioData.hasOwnProperty('category')){
+            dispatchOfProductsBeingShown({ type: "filterBySection", payload: eachVedioData.category })
+        }else
+            if(eachVedioData.hasOwnProperty('hasDiscount')){
+                dispatchOfProductsBeingShown({ type: "filterByProducts", payload: "hasDiscount" })
+            }else
+                if(eachVedioData.hasOwnProperty('onSale')){
```

Comment:
You can do this simply by eachVedioData['hasDiscount']

Reply:
thanks for suggesting, disn't notice so much of boilerplate code, will remove the fat from it asap!!